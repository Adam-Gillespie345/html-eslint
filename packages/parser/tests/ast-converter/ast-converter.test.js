"use strict";

const createASTConverter = require("../../lib/ast-converter/ast-converter");
const BASE = require("../fixtures/parse5/base.json");

describe("emitter", () => {
  let converter = null;

  beforeEach(() => {
    converter = createASTConverter();
  });

  test("The listener should not be called if the event name is different", () => {
    expect(converter.convert(BASE)).toMatchSnapshot();
  });
});
