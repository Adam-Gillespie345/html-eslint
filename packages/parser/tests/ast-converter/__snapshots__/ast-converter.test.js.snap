// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emitter The listener should not be called if the event name is different 1`] = `
Object {
  "body": Array [],
  "childNodes": Array [
    Object {
      "end": 15,
      "extra": Object {
        "name": "html",
        "publicId": "",
        "systemId": "",
      },
      "loc": Object {
        "end": Object {
          "column": 16,
          "line": 1,
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "range": Array [
        0,
        15,
      ],
      "start": 0,
      "type": "documentType",
    },
    Object {
      "childNodes": Array [
        Object {
          "endCol": 0,
          "endLine": 0,
          "endOffset": 0,
          "extra": Object {
            "attrs": Array [],
            "namespaceURI": "http://www.w3.org/1999/xhtml",
            "tagName": "head",
          },
          "startCol": 0,
          "startLine": 0,
          "startOffset": 0,
          "type": "head",
        },
        Object {
          "childNodes": Array [
            Object {
              "end": 36,
              "extra": Object {
                "value": "
    ",
              },
              "loc": Object {
                "end": Object {
                  "column": 5,
                  "line": 4,
                },
                "start": Object {
                  "column": 9,
                  "line": 3,
                },
              },
              "range": Array [
                31,
                36,
              ],
              "start": 31,
              "type": "text",
            },
            Object {
              "childNodes": Array [
                Object {
                  "end": 56,
                  "extra": Object {
                    "value": "My First Heading",
                  },
                  "loc": Object {
                    "end": Object {
                      "column": 25,
                      "line": 4,
                    },
                    "start": Object {
                      "column": 9,
                      "line": 4,
                    },
                  },
                  "range": Array [
                    40,
                    56,
                  ],
                  "start": 40,
                  "type": "text",
                },
              ],
              "end": 61,
              "extra": Object {
                "attrs": Array [],
                "namespaceURI": "http://www.w3.org/1999/xhtml",
                "tagName": "h1",
              },
              "loc": Object {
                "end": Object {
                  "column": 30,
                  "line": 4,
                },
                "start": Object {
                  "column": 5,
                  "line": 4,
                },
              },
              "range": Array [
                36,
                61,
              ],
              "start": 36,
              "type": "h1",
            },
            Object {
              "end": 64,
              "extra": Object {
                "value": "
  ",
              },
              "loc": Object {
                "end": Object {
                  "column": 3,
                  "line": 5,
                },
                "start": Object {
                  "column": 30,
                  "line": 4,
                },
              },
              "range": Array [
                61,
                64,
              ],
              "start": 61,
              "type": "text",
            },
            Object {
              "childNodes": Array [
                Object {
                  "end": 86,
                  "extra": Object {
                    "value": "My first paragraph.",
                  },
                  "loc": Object {
                    "end": Object {
                      "column": 25,
                      "line": 5,
                    },
                    "start": Object {
                      "column": 6,
                      "line": 5,
                    },
                  },
                  "range": Array [
                    67,
                    86,
                  ],
                  "start": 67,
                  "type": "text",
                },
                Object {
                  "end": 93,
                  "extra": Object {
                    "attrs": Array [],
                    "namespaceURI": "http://www.w3.org/1999/xhtml",
                    "tagName": "u",
                  },
                  "loc": Object {
                    "end": Object {
                      "column": 32,
                      "line": 5,
                    },
                    "start": Object {
                      "column": 25,
                      "line": 5,
                    },
                  },
                  "range": Array [
                    86,
                    93,
                  ],
                  "start": 86,
                  "type": "u",
                },
              ],
              "end": 97,
              "extra": Object {
                "attrs": Array [],
                "namespaceURI": "http://www.w3.org/1999/xhtml",
                "tagName": "p",
              },
              "loc": Object {
                "end": Object {
                  "column": 36,
                  "line": 5,
                },
                "start": Object {
                  "column": 3,
                  "line": 5,
                },
              },
              "range": Array [
                64,
                97,
              ],
              "start": 64,
              "type": "p",
            },
            Object {
              "end": 114,
              "extra": Object {
                "value": "


",
              },
              "loc": Object {
                "end": Object {
                  "column": 1,
                  "line": 8,
                },
                "start": Object {
                  "column": 36,
                  "line": 5,
                },
              },
              "range": Array [
                97,
                114,
              ],
              "start": 97,
              "type": "text",
            },
          ],
          "end": 114,
          "extra": Object {
            "attrs": Array [],
            "namespaceURI": "http://www.w3.org/1999/xhtml",
            "tagName": "body",
          },
          "loc": Object {
            "end": Object {
              "column": 1,
              "line": 8,
            },
            "start": Object {
              "column": 3,
              "line": 3,
            },
          },
          "range": Array [
            25,
            114,
          ],
          "start": 25,
          "type": "body",
        },
      ],
      "end": 114,
      "extra": Object {
        "attrs": Array [],
        "namespaceURI": "http://www.w3.org/1999/xhtml",
        "tagName": "html",
      },
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 8,
        },
        "start": Object {
          "column": 1,
          "line": 2,
        },
      },
      "range": Array [
        16,
        114,
      ],
      "start": 16,
      "type": "html",
    },
  ],
  "comments": Array [],
  "end": 114,
  "extra": Object {
    "mode": "no-quirks",
  },
  "loc": Object {
    "end": Object {
      "column": 1,
      "line": 8,
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "range": Array [
    0,
    114,
  ],
  "start": 0,
  "tokens": Array [],
  "type": "Program",
}
`;
